jenkinsci/foreman-node-sharing-plugin
# Presnet host detials on info page

In order to implement that, internal representation of a host had to change.

- `HostInfo` is now a snapshot of foreman host details preserved at Jenkins side.
  - All queries for individual attributes ware replaced by this one.
- `Cloud#hostsMap` represents possibly outdated hosts *inventory*. It is updated often enough to pick changes to the inventory itself (what nodes are there, their attributes) but the reservation status needs to be re-queried.

- Test updates
  - Use random wire mock ports so tests can be run in parallel
  - Support wire mock configuration from multiple threads - namely Jenkins build
  - Avoid large shared fixture
    - Rename files so it is clear what responses they represent
    - Remove replies no longer needed by the plugin
    - Provide fine grained stub methods to configure foreman state as desired on per-test basis
    - Add reusable scenarios for reservation/release

I wanted this to be a small change but I failed miserably.  Do not open the can of worms they said...
